modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -3603,6 +3603,15 @@ private void addVars(CodegenModel m, List<CodegenProperty> vars, Map<String, Sch
         return aliases;
     }
 
+    private static Boolean isAliasOfSimpleTypes(Schema schema) {
+        String schemaType = getPrimitiveType(schema);
+        return (schemaType != null
+                && !schemaType.equals("object")
+                && !schemaType.equals("array")
```

Comment:
What about using `ModelUtils.isObjectSchema` and `ModelUtils.isArraySchema` instead?

(ModeUtils was not available when I wrote these lines if I remember correctly)

Reply:
Thanks for your feedback! fixed it in https://github.com/OpenAPITools/openapi-generator/pull/2758/commits/23e4c4888685379e37929a4ddd4f6bc369795480